api/app/v1/lineage/LineageController.scala

```diff
@@ -100,7 +100,7 @@ class LineageController @Inject()(cc: LineageControllerComponents)
         // Produce a proper HTTP response for different success cases from LineageService
         lineage.transform {
           // A failure fetching lineage produces a 500 error code
-          case Failure(_) => Success(InternalServerError(s"""Unexpected error constructing lineage for table "$name""""))
```

Comment:
Negative depth is not supported. It is a validation error (bad request). With this change all server errors would return 404. We want to still return 500 for server errors .  We should handle it in block of code where we are validating queryParams.